homeassistant/components/conversation/default_agent.py

```diff
@@ -898,9 +949,14 @@ def _make_slot_lists(self) -> dict[str, SlotList]:
 
         self._slot_lists = {
             "area": TextSlotList.from_tuples(area_names, allow_template=False),
-            "name": TextSlotList.from_tuples(entity_names, allow_template=False),
+            "name": TextSlotList.from_tuples(
+                exposed_entity_names, allow_template=False
+            ),
             "floor": TextSlotList.from_tuples(floor_names, allow_template=False),
         }
+        self._all_names_list = TextSlotList.from_tuples(
```

Comment:
is this an expensive operation? Can we postpone the generation?

Reply:
It shouldn't be too expensive, so you're right we could postpone it until an error occurs.

Reply:
Delayed the creation of this list until needed, and also skipping config/hidden entities.